### ISSUE TITLE
Getting "unknown lint: `clippy::elidable_lifetime_names`" using clippy 1.85.1

### ISSUE THREAD

**Issue by corneliusroemer** [rust-lang/rust-clippy#14653]
### Summary

I'm getting an "unknown lint" warning for `clippy::elidable_lifetime_names` despite using rust/clippy version 1.85.1 which is later than the version `clippy::elidable_lifetime_names` was introduced in: 1.84.0 per https://rust-lang.github.io/rust-clippy/master/index.html#/elid)

Interestingly I couldn't find the lint in the stable branch, only on master. Is that why I'm getting the lint? If so, shouldn't the docs distinguish between the version something landed on stable?

### Lint Name

 unknown lint: `clippy::elidable_lifetime_names`

### Reproducer

I tried this code:

```rust
#![allow(
    clippy::elidable_lifetime_names,
)]

fn main() {
    println!("Hello, world!");
}
```

I saw this happen:

```
cargo clippy
    Checking tmp2 v0.1.0 (/Users/cr/code/tmp2)
warning: unknown lint: `clippy::elidable_lifetime_names`
 --> src/main.rs:3:5
  |
3 |     clippy::elidable_lifetime_names,
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: did you mean: `clippy::single_char_lifetime_names`
  |
  = note: `#[warn(unknown_lints)]` on by default

warning: `tmp2` (bin "tmp2") generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.06s

```

I expected to see this happen: No lint

### Version

```text
rustc 1.85.1 (4eb161250 2025-03-15)
binary: rustc
commit-hash: 4eb161250e340c8f48f66e2b929ef4a5bed7c181
commit-date: 2025-03-15
host: aarch64-apple-darwin
release: 1.85.1
LLVM version: 19.1.7
```

**Comment by corneliusroemer** [comment]
Some further wrong versions:
- `clippy::mem_replace_option_with_some`: should be 1.87 not 1.86
- `clippy::manual_dangling_ptr`: should be 1.88 not 1.87
- `clippy::char_indices_as_byte_indices`: should be 1.88 not 1.83
- `clippy::manual_abs_diff` should be 1.88 not 1.86
- `clippy::ignore_without_reason` should be 1.88 not 1.85

I found these by letting clippy 1.87 run on a file with all the declared lints (on master), looking for unknown lints and their official vs actual version of introduction.

**Comment by corneliusroemer** [comment]
Some more:
- `clippy:single_option_map` should be 1.87 not 1.86
- `clippy::owned_cow` should be 1.87 not 1.85
- `clippy::manual_contains` should be 1.87 not 1.86
- `clippy::manual_abs_diff` should be 1.87 not 1.86
- `clippy::doc_link_code` should be 1.87 not 1.86

**Comment by flip1995** [comment]
#14748 should've updated all of the ones that "should be 1.87".
